crates/turbopack-wasm/src/module_asset.rs

```diff
@@ -78,6 +79,12 @@ impl WebAssemblyModuleAsset {
             }))),
         ).module();
 
+        if let Some(esm_asset) =
+            Vc::try_resolve_downcast_type::<EcmascriptModulePartAsset>(module).await?
+        {
+            return Ok(esm_asset.await?.full_module);
+        }
+
```

Comment:
That doesn't seem right. We should just return the part asset. You should be able to widen the return value of this function to remove the cast below